narwhal/primary/src/tests/core_tests.rs

```diff
@@ -3,12 +3,67 @@
 // SPDX-License-Identifier: Apache-2.0
 use super::*;
 use crate::common::{create_db_stores, create_test_vote_store};
-use anemo::{types::PeerInfo, PeerId};
+use anemo::{async_trait, types::PeerInfo, PeerId};
 use fastcrypto::traits::KeyPair;
 use prometheus::Registry;
+use storage::ProposerStore;
 use test_utils::{fixture_batch_with_transactions, CommitteeFixture, PrimaryToPrimaryMockServer};
 use tokio::time::Duration;
-use types::{CertificateDigest, Header, Vote};
+use types::{
+    CertificateDigest, FetchCertificatesRequest, FetchCertificatesResponse, GetCertificatesRequest,
+    GetCertificatesResponse, Header, LatestHeaderResponse, PayloadAvailabilityRequest,
+    PayloadAvailabilityResponse, PrimaryToPrimary, Vote,
+};
+
+pub struct NetworkProxy {
+    proposer_store: ProposerStore,
+}
+
+#[async_trait]
+impl PrimaryToPrimary for NetworkProxy {
```

Comment:
instead of manually writing a mock implementation of the PrimaryToPrimary trait can you use the prebuilt automock version that comes with the code generation? it's called `MockPrimaryToPrimary`.

see `block_synchronizer_tests.rs:103` for an example of what this looks like. There are a few reasons handwritten mocks are not ideal:
1. it's much easier to follow the flow of what fake behavior the mock will perform when configured via expectations inline with the test
2. as you can see here, every we add a new method to the trait, all the other mocks in completely unrelated tests also need to be updated, which adds a lot of churn and boilerplate

Reply:
updated